app/models/tag.rb

```diff
@@ -48,6 +48,65 @@ def restricted?
     restricted
   end
 
+  # int -> Kaminari::PaginatableArray
+  def recent_edits_for_homepage(page = 1)
+    paginate page, recent_edits(page), (taggings.count * 2)
+  end
+
+  # -> [Hash]
+  # Adds the ActiveRecord Tagable with key 'tagable' to
+  # each hash provided by #recent_edits_query.
+  def recent_edits(page = 1)
+    edits = recent_edits_query(page)
+    active_record_lookup = active_record_lookup_for_recent_edits(edits)
+    edits.map do |edit|
+      edit.tap { |h| h.store 'tagable', active_record_lookup.dig(h['tagable_class'], h['tagable_id']) }
+    end
+  end
+
+  # int -> [ Hash ]
+  # Hash keys: tagging_id, tagable_id, tagable_class, tagging_created_at, event_timestamp, event
+  def recent_edits_query(page = 1)
+    sql = <<-SQL
+      (
+        SELECT taggings.id as tagging_id,
+                taggings.tagable_id,
+                taggings.tagable_class,
+                taggings.created_at as tagging_created_at,
+                COALESCE(entity.updated_at, ls_list.updated_at, relationship.updated_at) AS event_timestamp,
+	        'tagable_updated' as event
+	FROM taggings
+	LEFT JOIN entity ON taggings.tagable_id = entity.id AND taggings.tagable_class = 'Entity'
+ 	LEFT JOIN ls_list ON taggings.tagable_id = ls_list.id AND taggings.tagable_class = 'List'
+	LEFT JOIN relationship ON taggings.tagable_id = relationship.id AND taggings.tagable_class = 'Relationship'
+	WHERE taggings.tag_id = #{id}
+              # adding a tag triggers a callback that also updates the tagable. This clause excludes those updates
+	      AND abs(TIMESTAMPDIFF(SECOND, taggings.created_at, COALESCE(entity.updated_at, ls_list.updated_at, relationship.updated_at))) > 100
+      )
+        UNION
+      (
+        SELECT taggings.id as tagging_id,
+	       taggings.tagable_id,
+	       taggings.tagable_class,
+	       taggings.created_at AS tagging_created_at,
+	       taggings.created_at AS event_timestamp,
+	       'tag_added' AS event
+        FROM taggings
+	WHERE taggings.tag_id = #{id}
+      )
+        ORDER BY event_timestamp DESC
+        LIMIT #{TAGABLE_PAGINATION_LIMIT}
```

Comment:
`Tag::TAGABLE_PAGINATION_LIMIT` seems perhaps no longer accurately named?

if we intend to use 20 as the pagination limit for both tagable lists and edit lists, maybe lets just call it `Tag::PAGINATION_LIMIT` until we have a reason to have different paginations for different lists?

Reply:
Yeah, that makes sense. ``` <Model>::PAGINATION_LIMIT  ``` seems like a good convention to use.